File: algorithms/substract-and-conquer/binomial.ts
/**
 * Binomialkoeffizient
 *
 * Berechnet den Binomialkoeffizienten aus n und k.
 *
 * Komplexität: O(n^2)
 */

const binomial = (n: number, k: number): number => {
  if (k === 0 || k === n) {
    return 1;
  }
  return binomial(n - 1, k) + binomial(n - 1, k - 1);
};

const result = binomial(49, 6);
console.log(`Result: ${result}`);
export {};

File: algorithms/substract-and-conquer/binomial.py
"""
Binomialkoeffizient

Berechnet den Binomialkoeffizienten aus n und k.

Komplexität: O(n^2)
"""


def binomial(n: int, k: int) -> int:
    if n == k or k == 0:
        return 1
    return binomial(n - 1, k) + binomial(n-1, k-1)


if __name__ == "__main__":
    print(binomial(49, 6))
